0.ALL_DOWNLOAD.py

```diff
@@ -108,17 +127,17 @@ def read_avro_files(file_paths):
                 yield record
 
 
-def extract_fits_image(fits_bytes):
+def extract_and_process_image(fits_bytes, issdiffpos):
     """
     Extracts FITS image data from bytes, converts to float64, replaces NaNs with 0.
     """
-    import gzip
-    from astropy.io import fits
-    import io
     with gzip.open(io.BytesIO(fits_bytes), 'rb') as gz:
         with fits.open(io.BytesIO(gz.read())) as hdul:
-            image_data = hdul[0].data.astype(np.float32)
+            image_data = hdul[0].data.astype(np.float64)
```

Comment:
Why do you want it in float64? The original data is float32:
```
BITPIX  =                  -32
```
```suggestion
            image_data = hdul[0].data.astype(np.float32)
```